### DISCUSSION TITLE
Package extension for surfaceControl

### DISCUSSION THREAD

**Discussion by anubhav-dogra** [sam-xl/scancontrol#39] [Q&A]
Hi, can this package be extended for Micro Epsilon surfaceControl cameras?

**Comment by dave992** [comment]
As it is a different type of sensor, it would probably be in a separate repository. We currently do not have any surfaceCONTROL devices, so we would be unable to test the driver. I therefore think this is not something we will be working on in the near future (at least not until we have access to a surfaceCONTROL device). 

Does it use the same SDK (just a newer version?). If so the adaptation of the scanCONTROL driver is probably quite straightforward and I would suggest forking this repository. If it is a different SDK, it is probably more work.

**Comment by anubhav-dogra** [comment]
I think its a different one because the SDK for surface control is not available for linux platform. 
If we check this link, https://www.micro-epsilon.co.uk/2D_3D/3D-sensors/Softwareintegration/ , it suggests that SDK can be used for GenICam based camera, However with an additional requirement of the following, as mentioned in the documentation.
"In order to use the 3D-Sensor SDK, the Pleora eBus Runtime 6.1.9 has to be installed. This is done automatically when installing Micro-Epsilon 3D-View or 3DInspect."
and I think, these are not available for Linux platform !

**Comment by anubhav-dogra** [comment]
Hi again,

I got success in compiling the camera software within the ROS node and getting the sensor_msgs/Image or sensor_msgs/PointCloud2 topics. 
However, Im struggling to set the parameters for the camera_info topic. Can you suggest how to get these params ? Do I need to calibrate the camera first as given in this [link](http://wiki.ros.org/camera_calibration)  ? 
I am confused about the procedure.

**Comment by dave992** [comment]
Have a look at: https://docs.ros.org/en/noetic/api/sensor_msgs/html/msg/CameraInfo.html. It explains a little about the message and what to set (also in case it is uncalibrated). Calibration is needed if you want to retrieve the intrinsic and/or extrinsic calibration.

I am not sure if a `camera_info` message is needed in this particular case. If I understand correctly, it is just there to help convert raw images, as collected directly from the image sensor, to rectified images and inform the user about those. 

As this is a high-precision sensor, I assume the retrieved image is actually already a rectified image. I am not an expert on this topic, so please have a look yourself as well to confirm this. 

If it is already rectified, doing your own calibration on top might actually make the camera less accurate in this particular case as your calibration result can have a lower overall accuracy than the camera has by default (depending on your calibration method and tools).
